packages/react-instantsearch-hooks/src/__tests__/useQueryRules.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import { createInstantSearchTestWrapper } from '../../../../test/utils';
+import { useQueryRules } from '../useQueryRules';
+
+describe('useQueryRules', () => {
+  test('returns the connector render state', async () => {
+    const wrapper = createInstantSearchTestWrapper();
+
+    const trackedFilters = {
+      genre: () => ['Comedy', 'Thriller'],
+      rating: (values) => values,
+    };
+
+    const transformRuleContexts = (ruleContexts) =>
+      ruleContexts.map((ruleContext) => ruleContext.replace('ais-', 'custom-'));
+
+    const { result, waitForNextUpdate } = renderHook(
+      () =>
+        useQueryRules({
+          trackedFilters,
+          transformRuleContexts,
+        }),
+      {
+        wrapper,
+      }
+    );
+
+    // Initial render state from manual `getWidgetRenderState`
+    expect(result.current).toEqual({
+      items: [],
+    });
+
+    await waitForNextUpdate();
+
+    // InstantSearch.js state from the `render` lifecycle step
+    expect(result.current).toEqual({
+      items: [],
```

Comment:
We can inject some rules in the API response, otherwise this test doesn't reveal much.

Reply:
fix in b1e599bfde04a21be1178838bba0e0540c1d8acb